viadot/task_utils.py

```diff
@@ -668,3 +668,101 @@ def anonymize_df(
 
     df.drop(columns=["temp_date_col"], inplace=True, errors="ignore")
     return df
+
+
+import re
+
+
+@task(timeout=3600)
+def validate_df(df: pd.DataFrame, tests: dict = None) -> None:
+    """
+    Task to validate the data on DataFrame level.
+    tests:
+        - `column_size`: dict{column: size}
+        - `column_unique_values`: list[columns]
+        - `column_list_to_match`: list[columns]
+        - `dataset_row_count`: dict: {'min': number, 'max', number}
+        - `column_match_regex`: dict: {column: 'regex'}
+
+    Args:
+        df (pd.DataFrame): The data frame for validation.
+        tests (dict, optional): Tests to apply on the data frame. Defaults to None.
+
+    Raises:
+        ValidationError: If validation failed for at least one test.
+    """
+    failed_tests = 0
+    if tests is not None:
+        # column_row_count
+        if "column_size" in tests:
+            try:
+                for k, v in tests["column_size"].items():
+                    column_max_length = (
+                        df.astype(str).apply(lambda s: s.str.len()).max().to_dict()
```

Comment:
Change single letters variables to more readable ones - `for k, v` use `for key, val` .  For s maybe try to use `column` or `col`

Reply:
@adrian-wojcik please move the import to the import section